lib/types/array/methods/index.js

```diff
@@ -374,7 +374,15 @@ const methods = {
       if (val != null && utils.hasUserDefinedProperty(val, '$each')) {
         atomics.$push = val;
       } else {
-        atomics.$push.$each = atomics.$push.$each.concat(val);
+        if (val.length === 1) {
+          atomics.$push.$each.push(val[0]);
+        } else if (val.length < 10000) {
```

Comment:
maybe note why this magic number was chosen? (i guess it is because of a stack overflow error)